File: springbootjparest/src/main/java/at/eder/springbootjparest/models/Participation.java
package at.eder.springbootjparest.models;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;
import java.util.Set;

@Data
@Entity
@NoArgsConstructor
@SuperBuilder
public class Participation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private double price;
    private double paidAlready;
    private String comments;

    @ManyToOne
    private Event event;

    @ManyToOne
    private Company company;

    @ManyToOne
    private Responsible responsible;


    @ManyToMany
    private Set<TimeSlot> timeSlotOffers;

    @ManyToMany
    private Set<Department> interestedDepartments;
}

File: WTMobileApp/app/src/main/java/at/eder/wirtschaftstagmobileapp/models/Participation.kt
package at.eder.wirtschaftstagmobileapp.models

data class Participation(
    var id: Long?,
    var price: Double?,
    var paidAlready: Double?,
    var comments: String?,
    var event: Event?,
    var company: Company?,
    var responsible: User?,
    var timeSlotOffer: List<TimeSlot>?,
    var interestedDepartments: List<Department>?
) {
    override fun toString(): String = "$id - ${event?.label} <-- ${company?.name}"
}